recipes/redis-plus-plus/all/conanfile.py

```diff
@@ -36,27 +38,54 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
+    @property
+    def _has_build_async(self):
+        return tools.Version(self.version) >= "1.3.0"
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if not self._has_build_async:
+            del self.options.enable_async
+
+    @property
+    def _min_cppstd_required(self):
+        return 17 if self._has_build_async else 11
+
+    @property
+    def _min_compilers_version(self):
+        return {
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10",
+            "Visual Studio": "15",
+        }
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
+            tools.check_min_cppstd(self, self._min_cppstd_required)
+        
+        if tools.Version(self.version) >= "1.3.0":
+            min_compiler_version = self._min_compilers_version.get(str(self.settings.compiler), False)
+            if min_compiler_version:
+                if tools.Version(self.settings.compiler.version) < min_compiler_version:
+                    raise ConanInvalidConfiguration("redis-plus-plus requires C++17, which your compiler does not support.")
+            else:
+                self.output.warn("redis-plus-plus requires C++17. Your compiler is unknown. Assuming it supports C++17.")
```

Comment:
seems quite excessive it async is disabled, since C++11 is sufficient according to `_min_cppstd_required` you introduced.

Reply:
Even disabled it required C++17 my naming was a little weak on this one